dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/dbms/HibernateDbmsManager.java

```diff
@@ -467,4 +469,16 @@ private void dropTable( String table )
             log.debug( "Table " + table + " does not exist" );
         }
     }
+
+    private void emptyRelationships()
+    {
+        try
+        {
+            jdbcTemplate.update( "update relationshipitem set relationshipid = null; delete from relationship; delete from relationshipitem" );

```

Comment:
Just wondering, what is the reason for the custom exception handling for relationship?

Reply:
We have foreign keys between relationship and relationship item (both ways) so just emptying one table and then the other will cause an error, regardless of which order we do it. so Instead we need to set the reference to null first and then empty them. I made it into its own special query so to try and keep it clean

Reply:
Okay got it.